Make use_c10_dispatcher: full mandatory for structured kernels

ezyang opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #49505 Add at::cpu namespace of functions for structured kernels
* #49498 Push anonymous namespace into codegen, not template
* **#49490 Make use_c10_dispatcher: full mandatory for structured kernels**
* #49489 Move default or no default logic into native.argument
* #49348 codegen: Resolve overload ambiguities created by defaulted arguments
* #49341 Tighten up error checking on manual_kernel_registration
* #49245 Construct CppSignatureGroup from NativeFunction

No reason to let people to do the legacy thing for the brand new kernel.
This simplifies the codegen.  I have to port the two structured kernels
to this new format.

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

Differential Revision: [D25595406](https://our.internmc.facebook.com/intern/diff/D25595406)

ezyang committed:
Make use_c10_dispatcher: full mandatory for structured kernels

No reason to let people to do the legacy thing for the brand new kernel.
This simplifies the codegen.  I have to port the two structured kernels
to this new format.

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

[ghstack-poisoned]

ezyang committed:
Update on "Make use_c10_dispatcher: full mandatory for structured kernels"


No reason to let people to do the legacy thing for the brand new kernel.
This simplifies the codegen.  I have to port the two structured kernels
to this new format.

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

Differential Revision: [D25595406](https://our.internmc.facebook.com/intern/diff/D25595406)

[ghstack-poisoned]

ezyang committed:
Update on "Make use_c10_dispatcher: full mandatory for structured kernels"


No reason to let people to do the legacy thing for the brand new kernel.
This simplifies the codegen.  I have to port the two structured kernels
to this new format.

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

Differential Revision: [D25595406](https://our.internmc.facebook.com/intern/diff/D25595406)

[ghstack-poisoned]

ezyang committed:
Update on "Make use_c10_dispatcher: full mandatory for structured kernels"


No reason to let people to do the legacy thing for the brand new kernel.
This simplifies the codegen.  I have to port the two structured kernels
to this new format.

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

Differential Revision: [D25595406](https://our.internmc.facebook.com/intern/diff/D25595406)

[ghstack-poisoned]

ezyang committed:
Update on "Make use_c10_dispatcher: full mandatory for structured kernels"


No reason to let people to do the legacy thing for the brand new kernel.
This simplifies the codegen.  I have to port the two structured kernels
to this new format.

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

Differential Revision: [D25595406](https://our.internmc.facebook.com/intern/diff/D25595406)

[ghstack-poisoned]

ezyang committed:
Update on "Make use_c10_dispatcher: full mandatory for structured kernels"


No reason to let people to do the legacy thing for the brand new kernel.
This simplifies the codegen.  I have to port the two structured kernels
to this new format.

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

Differential Revision: [D25595406](https://our.internmc.facebook.com/intern/diff/D25595406)

[ghstack-poisoned]

smessmer reviewed (approved):
note that I have a PR enforcing this for all ops at https://github.com/pytorch/pytorch/pull/49164

ezyang committed:
Update on "Make use_c10_dispatcher: full mandatory for structured kernels"


No reason to let people to do the legacy thing for the brand new kernel.
This simplifies the codegen.  I have to port the two structured kernels
to this new format.

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

Differential Revision: [D25595406](https://our.internmc.facebook.com/intern/diff/D25595406)

[ghstack-poisoned]

ezyang committed:
Update on "Make use_c10_dispatcher: full mandatory for structured kernels"


No reason to let people to do the legacy thing for the brand new kernel.
This simplifies the codegen.  I have to port the two structured kernels
to this new format.

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

Differential Revision: [D25595406](https://our.internmc.facebook.com/intern/diff/D25595406)

[ghstack-poisoned]
